examples/sphere.py

```diff
@@ -427,60 +811,24 @@ def sphere_main(algorithm,
         seed (int): Seed for the algorithm. By default, there is no seed.
     """
```

Comment:
Right now, I find the behavior a bit hard to maintain because you retrieve `CONFIG[algorithm]` here and also in `create_scheduler`. I think it would be better if you just retrieve it here (make sure to do a `copy.deepcopy`) and then override parameters as appropriate based on what is passed into this main function. Then you can pass this entire config object into `create_scheduler`. You can also pass it around in any subsequen functions. This way, you only deal with a single config object.